lang/java/avro/src/test/java/org/apache/avro/reflect/TestReflect.java

```diff
@@ -1048,4 +1048,19 @@ public void testAvroDefault() {
           +"{\"name\":\"foo\",\"type\":\"int\",\"default\":1}]}");
   }
 
+  public static class NullableBytesTest {
+    @Nullable
+    byte[] bytes = "foo".getBytes();
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof NullableBytesTest
+              && Arrays.equals(((NullableBytesTest) obj).bytes, this.bytes);
+    }
+  }
+
+  @Test
+  public void testNullableByteArray() throws Exception {
+    checkReadWrite(new NullableBytesTest());
```

Comment:
As we are testing a nullable field I would also test the scenario when this field is really null.